### DISCUSSION TITLE
separating vertical and horizontal grain boundaries

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#583] [Ask Anything] [answered]
Hello,

I am trying to separate between horizontal and vertical boundaries of the grains shown in the figure below.
![large_grains](https://user-images.githubusercontent.com/73701277/99779409-4eb93400-2b15-11eb-8c92-0fc5548ae8a4.jpg)

So I have written the following script:
```matlab
m = 1;
n = 1;
clear D horizontal_axis horizontal_angle vertical_axis vertical_angle horizontal_index vertical_index
for k = 1:length(large_grains)
D=angle(large_grains(k).boundary.direction,vector3d.X)/degree;
for i = 1:length(D)
 if D(i)== 0
    horizontal_axis(m)=large_grains(k).boundary.misorientation(i).axis;
    horizontal_angle(m)=large_grains(k).boundary.misorientation(i).angle./degree;
    horizontal_index(m)=i;
    m  = m + 1;
 else
    vertical_axis(n)=large_grains(k).boundary.misorientation(i).axis;
    vertical_angle(n)=large_grains(k).boundary.misorientation(i).angle./degree;
    vertical_index(n)=i;
    n  = n + 1;
 end
end
end 
```

Unfortunately, when I tried to plot the horizontal part of the boundary of one grain, I found that even the horizontal segments of the vertical boundaries are taken into account as seen in the figure below . 
![horizontal_index](https://user-images.githubusercontent.com/73701277/99780240-7f4d9d80-2b16-11eb-96a9-bbcce215ec19.jpg)

Actually, by horizontal boundaries, I mean the segments that are shown in the figure below 
![horizontal_boundaries](https://user-images.githubusercontent.com/73701277/99780972-8923d080-2b17-11eb-8221-44edee5d794d.JPG)

Does anyone know how to correct my script to get only the horizontal boundaries (as shown in the figure above) or vertical ones please?

Thank you a lot,

Saoussen

**Comment by ralfHielscher** [comment]
The misorientation axis / angle of a grain boundary refers to the misorientation with the neighboring grain and has nothing to do with the direction of the grain boundary segment.

What you want to check is
```matlab
angle(grains.boundary.direction, vector3d.X) == 0
```
Ralf.

**Reply by Saoussen-ouhiba** [reply]
Thany you for your reply. Yes, I have used the same function that you mentioned (see attached script) but it is not working because even the vertical boundaries contain some horizontal segments. 

```
for k = 1:length(large_grains)
D=angle(large_grains(k).boundary.direction,vector3d.X)/degree;
for i = 1:length(D)
 if D(i)== 0
    horizontal_axis(m)=large_grains(k).boundary.misorientation(i).axis;
    horizontal_angle(m)=large_grains(k).boundary.misorientation(i).angle./degree;
    horizontal_index(m)=i;
    m  = m + 1;
 else
    vertical_axis(n)=large_grains(k).boundary.misorientation(i).axis;
    vertical_angle(n)=large_grains(k).boundary.misorientation(i).angle./degree;
    vertical_index(n)=i;
    n  = n + 1;
 end
end
end 
```

**Comment by kilir** [comment, marked as answer]
Hi,
so by horizontal and vertical grain boundary segments, you mean some sort of average grain boundary trend, not the direction of the segments, right?
You can smooth grains and decide for some threshold:

```
mtexdata titanium

grains  =ebsd.calcGrains;
grains = grains.smooth(10);

gbd = grains.boundary.direction;

cond1 = angle(gbd,xvector)<=45*degree;

plot(grains.boundary(cond1),'linecolor','r')
hold on
plot(grains.boundary(~cond1),'linecolor','b')
hold off

```
Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you so much. It is exatly what I was looking for.
